autorest/preparer.go

```diff
@@ -197,6 +199,64 @@ func WithFormData(v url.Values) PrepareDecorator {
 	}
 }
 
+// WithMultiPartFormData returns a PrepareDecoratore that "URL encodes" (e.g., bar=baz&foo=quux) form parameters
+// into the http.Request body.
+func WithMultiPartFormData(formDataParameters map[string]interface{}) PrepareDecorator {
+	return func(p Preparer) Preparer {
+		return PreparerFunc(func(r *http.Request) (*http.Request, error) {
+			r, err := p.Prepare(r)
+			if err == nil {
+				body := &bytes.Buffer{}
```

Comment:
initialize as `var body bytes.Buffer` and parse as `&body` where you need to. it's more idiomatic.